src/common/test-webserver.c

```diff
@@ -803,19 +816,23 @@ test_address (TestCase *tc,
     }
   else
     {
-      assert_cannot_connect (tc->localport);
+      if (tc->hostport)
+        assert_cannot_connect (tc->localport);
```

Comment:
This may look a bit weird. If there is only one interface, then `cockpit_web_server_new` will get a NULL address and thus *do* listen on loopback (as there is no other place to listen). So this is intended.